index.html

```diff
@@ -334,18 +334,12 @@
 	<label>Type an email: <input type="email"></label>
 	<pre class="language-markup"><code>&lt;input type="email" /></code></pre>
 	<pre class="language-javascript"><code><script>new Awesomplete($('input[type="email"]'), {
-	list: ["@aol.com", "@att.net", "@comcast.net", "@facebook.com", "@gmail.com", "@gmx.com", "@googlemail.com", "@google.com", "@hotmail.com", "@hotmail.co.uk", "@mac.com", "@me.com", "@mail.com", "@msn.com", "@live.com", "@sbcglobal.net", "@verizon.net", "@yahoo.com", "@yahoo.co.uk"],
-	item: function(text, input) {
-		var newText = input.slice(0, input.indexOf("@")) + text;
-
-		return Awesomplete.$.create("li", {
-			innerHTML: newText.replace(RegExp(input.trim(), "gi"), "<mark>$&</mark>"),
-			"aria-selected": "false"
-		});
+	list: ["aol.com", "att.net", "comcast.net", "facebook.com", "gmail.com", "gmx.com", "googlemail.com", "google.com", "hotmail.com", "hotmail.co.uk", "mac.com", "me.com", "mail.com", "msn.com", "live.com", "sbcglobal.net"],
+	data: function (text, input) {
```

Comment:
Now the `text` is an item from the list in original format. In this case it's a string.

So we work with string here and return string, as it's wrapped in `Suggestion` after that internally, which supports item being String, Array of Strings, Object with `label` and `value` String properties.

So if you want to support any other type of items, say Date, you need to return an item from `data` method in some of the supported by default String/Array/Object formats.

Example for date completer:

```javascript
// same label/value - string representation of Date
list: [ new Date("2015-01-01"), new Date("2015-01-02"), ... ]
data: function (item, input) {
	return item.toString();
}

// for different label/value - label is a string representation of Date, value is a timestamp
list: [ new Date("2015-01-01"), new Date("2015-01-02"), ... ]
data: function (item, input) {
	return { label: item.toString(), value: item.valueOf() };
}
```